clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -594,7 +635,7 @@ private void handleShareFetchSuccess(Node fetchTarget,
                     topicResponse.partitions().forEach(partition ->
                             responseData.put(new TopicIdPartition(topicResponse.topicId(),
                                     partition.partitionIndex(),
-                                    metadata.topicNames().get(topicResponse.topicId())), partition)));
+                                    metadata.topicNames().getOrDefault(topicResponse.topicId(), forgottenTopicNames.remove(topicResponse.topicId()))), partition)));
```

Comment:
This seems wrong to me. Each partition for a forgotten topic will cause `forgottenTopicNames.remove()` to be called, and only the first call will be valid. The handling of the forgotten topic names needs to be a bit less clever.

Reply:
Right, thanks for catching this. I have now used `partitionIndex` along with `topicId` as the key for the map to ensure we have removed only the entry corresponding to that partition.